Snowflake.Data.Tests/SFConnectionPoolT.cs

```diff
@@ -105,6 +108,36 @@ public void TestConnectionPoolIsFull()
         }
 
         [Test]
+        public void TestConnectionPoolExpirationWorks()
+        {
+            System.Threading.Thread.Sleep(3000); // wait for 3 seconds, in case other test still running.
+            SnowflakeDbConnectionPool.ClearAllPools();
+            SnowflakeDbConnectionPool.SetMaxPoolSize(2);
+            SnowflakeDbConnectionPool.SetTimeout(10);
+            SnowflakeDbConnectionPool.SetPooling(true);
+
+            var conn1 = new SnowflakeDbConnection();
+            conn1.ConnectionString = ConnectionString;
+
+            conn1.Open();
+            conn1.Close();
+            SnowflakeDbConnectionPool.SetTimeout(-1);
+
+            var conn2 = new SnowflakeDbConnection();
+            conn2.ConnectionString = ConnectionString;
+            conn2.Open();
+            conn2.Close();
+            var conn3 = new SnowflakeDbConnection();
+            conn3.ConnectionString = ConnectionString;
+            conn3.Open();
+            conn3.Close();
+
+            Assert.AreEqual(1, SnowflakeDbConnectionPool.GetCurrentPoolSize());
```

Comment:
Why do you expect 1 here?

Reply:
Since those connections use the same connection string. please look at line 120, 127, 131. So the same connection will reuse from the pool. And this test case is provided by the customer.